src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinGuiTextField.java

```diff
@@ -0,0 +1,58 @@
+package com.mitchej123.hodgepodge.mixins.early.minecraft;
+
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.gui.GuiTextField;
+
+import org.lwjgl.input.Keyboard;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(GuiTextField.class)
+public abstract class MixinGuiTextField {
+
+    @Shadow
+    private boolean isEnabled;
+
+    @Shadow
+    private boolean isFocused;
+
+    @Shadow
+    public abstract void writeText(String p_146191_1_);
+
+    @Shadow
+    public abstract void setCursorPositionEnd();
+
+    @Shadow
+    public abstract void setSelectionPos(int p_146199_1_);
+
+    @Shadow
+    public abstract String getSelectedText();
+
+    @Inject(method = "textboxKeyTyped", at = @At(value = "HEAD"), cancellable = true)
+    public void provideMacOsKeys(char typedChar, int eventKey, CallbackInfoReturnable<Boolean> cir) {
```

Comment:
And make it private as well, sry I keep adding comments

Reply:
Also do setReturnValue works as return in case if you have condition return?

Do this 
`if (value) {
    return 1
        }
return 2`
Works like this
`if (value) {
    setReturnValue(1)
        }
setReturnValue(2)`

Or you still need else?

Reply:
I advise you to look at the decompiled output of mixins to get an answer to your question. When running the client in dev environment with mixins, the classes transformed by mixins are dumped in a folder and you can look at what your mixin does to the code. The folder is in run/.mixin_out